evolution/fitness.py

```
@@ -633,39 +687,51 @@ def _apply_composite_logic(self, logic: str, scores: Dict[str, float]) -> float:
         # This is a simplified implementation - in production you'd want a proper expression parser
         # Replace function names with their scores
         safe_expr = expression
         for name, score in scores.items():
             safe_expr = safe_expr.replace(name, str(score))
-        
+
         # Only allow safe operations
-        allowed_chars = set('0123456789.+-*/ ()')
+        allowed_chars = set("0123456789.+-*/ ()")
         if not all(c in allowed_chars for c in safe_expr):
             raise ValueError(f"Unsafe expression: {expression}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)